### PR TITLE
apm collector ui module testcase completed

### PR DESCRIPTION
**Pull request by candyleer**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [ ] Improve performance
- [x] Test Case

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@peng-yongsheng This PR still depends on your preview.

**Comment by peng-yongsheng** [comment]
@wu-sheng I'm reviewing automated test now, it's too late. I will review this tomorrow morning.

File: apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/test/java/org/apache/skywalking/apm/collector/ui/service/TraceStackServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        globalTraceDAO = mock(IGlobalTraceUIDAO.class);
+        segmentDAO = mock(ISegmentUIDAO.class);
+        applicationCacheService = mock(ApplicationCacheService.class);
+        Whitebox.setInternalState(traceStackService, "globalTraceDAO", globalTraceDAO);
+        Whitebox.setInternalState(traceStackService, "segmentDAO", segmentDAO);
+        Whitebox.setInternalState(traceStackService, "applicationCacheService", applicationCacheService);
+    }
+
+    @Test
+    public void load() {
```
Comment (peng-yongsheng) [inline apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/test/java/org/apache/skywalking/apm/collector/ui/service/TraceStackServiceTest.java]: This case did't match real scenario, some key logic not test.

1. Global trace with the id equal to `1` contains 3 segments(e.g., A, B, C), disorder their.
1. Call relation: A -> B,  A -> C, so entry span in B and C have reference information.
1. Verify the order of spans.

File: apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/test/java/org/apache/skywalking/apm/collector/ui/service/TopologyBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        duration.setStart("2018-01");
+        duration.setStep(Step.MONTH);
+        startSecondTimeBucket = DurationUtils.INSTANCE.startTimeDurationToSecondTimeBucket(duration.getStep(), duration.getStart());
+        endSecondTimeBucket = DurationUtils.INSTANCE.endTimeDurationToSecondTimeBucket(duration.getStep(), duration.getEnd());
+        startTimeBucket = DurationUtils.INSTANCE.exchangeToTimeBucket(duration.getStart());
+        endTimeBucket = DurationUtils.INSTANCE.exchangeToTimeBucket(duration.getEnd());
+    }
+
+    @Test
+    public void build() throws ParseException {
```
Comment (peng-yongsheng) [inline apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/test/java/org/apache/skywalking/apm/collector/ui/service/TopologyBuilderTest.java]: @wu-sheng Real scenario of this test is very hard to mock. I suggest use automated test with the real segment data to test this method. So, this test case is enough.

Comment (wu-sheng) [reply apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/test/java/org/apache/skywalking/apm/collector/ui/service/TopologyBuilderTest.java]: @peng-yongsheng You can feel relax about these type test cases. I agree, on one can be easy to mock topology and trace related test cases. We can count on your automatic backend test to cover these.

File: apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/test/java/org/apache/skywalking/apm/collector/ui/service/TraceStackServiceTest.java (hunk 1 above)
Comment (wu-sheng) [reply apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/test/java/org/apache/skywalking/apm/collector/ui/service/TraceStackServiceTest.java]: `TraceStackService` will be changed totally by me, today. So I prefer to merge these test cases before that.

**Comment by candyleer** [comment]
yeah ,some cases  i found it's hard to write meaningful test case, so right now i just  cover the target code, i can optimize this in the future when I become more familiar with code and design ideas.

**Pull request merged**